source/mir/random/ndvariable.d

```diff
@@ -279,6 +280,105 @@ nothrow @safe version(mir_random_test) unittest
     assert(fabs(x[0] + x[1] + x[2] - 1) < 1e-10);
 }
 
+/++
+Multinomial distribution.
++/
+struct MultinomialVariable(T)
+    if (isFloatingPoint!T)
+{
+    import mir.random.variable : binomialVar;
+
+    ///
+    enum isNdRandomVariable = true;
+    ///
+    alias Element = uint;
+
+    ///
+    const(T)[] probs;
+    size_t N;
+    double norm = 0.0;
```

Comment:
should be removed

Reply:
If I remove this I get this error:
```D
source/mir/random/ndvariable.d(378,7): Error: template mir.random.ndvariable.MultinomialVariable!double.MultinomialVariable.opCall cannot deduce function from argument types !()(MersenneTwisterEngine!(ulong, 64LU, 312LU, 156LU, 31LU, 13043109905998158313LU, 29LU, 6148914691236517205LU, 17LU, 8202884508482404352LU, 37LU, 18444473444759240704LU, 43LU, 6364136223846793005LU), ulong[]), candidates are:
source/mir/random/ndvariable.d(327,10):        mir.random.ndvariable.MultinomialVariable!double.MultinomialVariable.opCall(G)(ref scope G gen, scope uint[] result) if (isSaturatedRandomEngine!G)
source/mir/random/ndvariable.d(353,10):        mir.random.ndvariable.MultinomialVariable!double.MultinomialVariable.opCall(G)(scope G* gen, scope uint[] result) if (isSaturatedRandomEngine!G)
/usr/bin/dmd failed with exit code 1.
```

Reply:
well, then take a look into the function in the error msg and try to get what is going on here
should not be a big problem

Reply:
These variables need to be declared here because they have their values set in the constructor and are used later in the `opCall`. Other variables in this module use the same pattern. For example `DirichletVariable` does this for its `alpha` parameter.